### DISCUSSION TITLE
Unwanted update for old Macs (Firefox)

### DISCUSSION THREAD

**Discussion by jcfrog** [aclap-dev/video-downloadhelper#1] [Announcements]
You may have received an update requiring a new CoApp that doesn't work for your operating system (< 10.15).

We are aware of the issue and are working on a fix.

In the meantime, if you want to go back, the previous versions are here:

- Extension: https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/
- CoApp: https://www.downloadhelper.net/install-coapp-v1

Please configure your extension so that it is not automatically updated.

**Comment by ericmarceau-rogers** [comment]
Respectfully, for your consideration (_from the earlier posting_):

- https://groups.google.com/g/video-downloadhelper-q-and-a/c/e4EstntVB8s/m/tMqxWM69AgAJ

    [Firefox]



**[1] Plug-in Level**

The issue with these types of problems (**Browser updates**) is that the Browser's **default setting** for **auto-update** of Plug-ins is "Auto", meaning "ON".

That is an issue which **may** be out of VDH's control.  However, the page at

https://extensionworkshop.com/documentation/manage/updating-your-extension/

**seems to suggest** that the _Plug-in's auto-update settings are_ **config-file** _controllable by the Developer_.  So I leave that to them to review and, if need be, reconsider setting the default setting for auto-updates to "false", rather than "true".  IMHO, "false" seems to be the mode that "protects" the end-user and advise that this should be the default.  This in turn would put the onus for ALL auto-update action _completely_ into the individual user's hands, and control and prevent all future "surprise" updates having issues at times that disrupt other user priorities.

It would be nice if developers could review their earlier/current stance,
decide on a stance for the application going forward, and
communicate to the community what that stance would be.  (hopefully a "false" default 🙂)



**[2] Browser Level**

I've tried to find a Browser global setting (Firefox) to set the "auto-update" action default as "OFF".

For Firefox,  using

   **about:config**                [**NOTE**:  proceed with caution]

and searching for "**update**", I found the settings

   **app.update.auto**                                        true   [_browser default_]
      http://kb.mozillazine.org/App.update.auto
      https://support.mozilla.org/en-US/questions/1315028
         _Seems to suggest applicable to Browser, not plug-ins_

   **extensions.update.autoUpdateDefault**      true   [_browser default_]
      https://support.mozilla.org/en-US/questions/952162
         _Will auto-update **ONLY IF**_    **extensions.update.enabled**     _has identified an update is available_

   **extensions.update.enabled**                       true   [_browser default_]
      https://blog.mozilla.org/addons/how-to-turn-off-add-on-updates/
         _Global control switch on whether to attempt any "Add-on update scan" for update availability.
         "**false**" suppresses all globally
         (I **don't know** if that would suppress the choice radio-buttons for each individual plug-in's settings.)_

So, this last setting seems to be the one to modify for a global freeze on plug-ins.  I am still having an internal debate on whether I want to go that route or to stick with control at the plug-in level.  After all, if this last one is set false, then later set to true, ALL the plug-ins would be updated at the same time, possibly creating an impossible nightmare if each individual plug-in had similar update issues.  I don't think I would want to contemplate that, as a desktop user.  

However, from a standpoint of an Admin for a group of users, that would definitely be attractive to ensure a "freeze" on deployed infrastructure that would ensure business continuity.   So, the course to follow very much depends on your context.


Given all that, from an individual Desktop User's standpoint, **unless** you set  "extensions.update.enabled"  to  "**false**", the issue is one of having to "repair" plug-ins after each Browser update, which dictates a User/Admin control on when and how the Browser itself should be updated, and be prepared with a set of actions for "cleanup" after each such update.  That would include manual change of the auto-update setting for ALL plug-ins to "OFF" manually. 🙁

Food for thought!


Eric
just a user_

**Comment by GrampaWildWilly** [comment]
I think a discussion of how a user manages his other add-ons is a bit off-topic here.  But for VDH, I agree that auto-updates need to be disabled.  This is because it appears that when the browser decides it will auto-update VDH, it does it, without checking whether the user is actually in the process of using VDH to download something.  I must emphasize again, as I have elsewhere, it is the browser that decides to do the update, not VDH.  A couple of users reported this happening to them, & they were quite justifiably very upset.  So if there is any way for VDH to control whether VDH -- not any other add-on, just VDH -- gets auto-updated, it is an utter no-brainer that VDH should turn this browser setting for automatic updates off.  At the very least, it should be explained that you should disable auto-updates for VDH & the reason should be clearly explained.  People aren't going to accept this advice without an explanation so the explanation is of prime importance.  This should be placed somewhere it would be impossible for a new user to miss.

**Comment by jadenquinn** [comment]
> You may have received an update requiring a new CoApp that doesn't work for your operating system (< 10.15).
> 
> We are aware of the issue and are working on a fix.
> 
> In the meantime, if you want to go back, the previous versions are here:
> 
>     * Extension: https://addons.mozilla.org/en-US/firefox/addon/video-downloadhelper/versions/
> 
>     * CoApp: https://www.downloadhelper.net/install-coapp-v1
> 
> 
> Please configure your extension so that it is not automatically updated.

The extension isn't recognizing that the coapp is installed. This doesn't work on 10.13.

**Comment by kit529738** [comment]
Hi-- I'm using high sierra on a system I cannot update past 10.13, and i've been able to use dl helper fine, using an older version of both the app and extension .[ Please view my explanation here ](https://github.com/aclap-dev/video-downloadhelper/discussions/288#discussioncomment-8966571)
